src/Models/Car/Car.js

```diff
@@ -0,0 +1,60 @@
+import {
+    CarNameNotStringError,
+    CarNameEmptyError,
+    CarNameTooLongError,
+} from './errors.js';
+
+export default class Car {
+    #name;
+    #position;
+
+    static INITIAL_POSITION = 0;
+    static NAME_MAX_LENGTH = 5;
+    static MOVE_STEP = 1;
+
+    static of(name, position = Car.INITIAL_POSITION) {
+        return new Car(name, position);
+    }
+
+    constructor(name, position = Car.INITIAL_POSITION) {
+        this.#validateName(name);
+
+        this.#name = name.trim();
+        this.#position = position;
+    }
+
+    #isNotString(name) {
+        return typeof name !== 'string';
+    }
+
+    #isEmpty(name) {
+        return name.trim().length === 0;
+    }
+
+    #isLongName(name) {
+        return name.trim().length > Car.NAME_MAX_LENGTH;
+    }
+
+    #validateName(name) {
+        if (this.#isNotString(name)) throw new CarNameNotStringError();
+        if (this.#isEmpty(name)) throw new CarNameEmptyError();
+        if (this.#isLongName(name)) throw new CarNameTooLongError();
+    }
+
+    getRecord() {
+        return {
+            name: this.#name,
+            position: this.#position,
+        };
+    }
```

Comment:
아예 getter로 만드는 것도 괜찮을 것 같아요!
```suggestion
    get record() {
        return {
            name: this.#name,
            position: this.#position,
        };
    }
```

Reply:
JS 클래스 문법에서 지원하는 getter를 사용하면 가독성이 좋긴하지만, 여기서는 객체 지향적 설계 방식을 추구하기 위해 메소드로 구현했습니다. 

getter를 사용하면 객체 내부 데이터가 외부로 노출된다는 점에서 데메테르 법칙을 위반한다고 생각했기 때문입니다.

getter를 사용하면: 
- `캡슐화 약화`: 객체의 데이터를 외부에 공개하기 때문에, 데이터가 외부에서 어떻게 사용될지 객체가 통제 불가
- `변경으로 인한 파급 효과 발생`: getter로 데이터를 외부에 노출하면, 외부 코드가 그 데이터를 직접 활용할 가능성이 높음 -> 의존성으로 인해 객체 내부 데이터 구조가 변경되면 외부 코드도 영향을 받아 수정해야 하는 부분이 많아질 수 있음

Reply:
멘토링 시간에 말씀해주신 것처럼, 

getter 사용시 장점:
- 객체 내부 코드 간결성
- 객체 외부에서 데이터 접근 방식의 간결성

**가독성 측면의 장점**을 고려해 리팩토링 여부를 결정해 보겠습니다.

Reply:
이 부분은 getter로 만들어 보니, 코드가 너무 객체 내부 구현에 종속되어 보여 다시 get[자료명] 형태의 메소드로 변경했습니다.

Reply:
멘토링 시간에 말씀해 주신 지점들을 고려해서 getter를 사용하는 형태로 코드 전체를 수정했습니다. 

💡 단, `getter 사용 기준: class 내 특정 상태 단순 반환`
💡 그 외의 경우(예_데이터에 특정 작업을 수행한 뒤 반환하는 등...)는 `get[자료명]` 형태의 함수 네이밍 컨벤션을 적용하였습니다.